graph/dictionary/index.md

```diff
@@ -0,0 +1,187 @@
+
+More information:
+
+- [OData reference](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#dictionary)
+
+## When to use dictionary types
+
+Before using a dictionary type in your API definition make sure your scenario fits the following criteria:
+
+- The data values MUST be related to one another semantically as a collection.
+- The value types MUST be a primitive type or is a **ComplexType**. Mixed primitive types are not allowed.
+- The client MUST define the keys of this type. As opposed to the service defining it in advance.
+
+### Alternatives to consider
+
+- [Open extensions](https://docs.microsoft.com/en-us/graph/extensibility-open-users) when you want to provide clients the ability to extend Microsoft Graph.
+- [Open types](https://docs.microsoft.com/en-us/aspnet/web-api/overview/odata-support-in-aspnet-web-api/odata-v4/use-open-types-in-odata-v4) when your data is not a collection in nature.
+- [Complex types](https://docs.microsoft.com/en-us/odata/webapi/complextypewithnavigationproperty) when the set of data values are known.
+
+## JSON payload example
+
+The following example illustrates the resulting JSON for a property of dictionary type. The parent object has been ommitted for brievety.
+
+```json
+{
+  "author": {
+    "domain": "contoso"
+  },
+  "maintainer": {
+    "domain": "fabrikam"
+  },
+  "architect": {
+    "domain": "adventureWorks"
+  }
+}
+```
+
+## HTTP calls examples
+
+In this set of examples we're modeling a **roles** property of dictionary type on the user entity which is exposed by the users entity set.
+
+### Getting an entry from the dictionary
+
+```HTTP
+GET https://graph.microsoft.com/v1.0/users/10/roles/author
+```
+
+Reponse:
+
+```json
+{
+  "domain": "contoso"
+}
+```
+
+### Getting the dictionary
+
+```HTTP
+GET https://graph.microsoft.com/v1.0/users/10/roles
+```
+
+Reponse:
+
+```json
+{
+  "author": {
+    "domain": "contoso"
+  },
+  "maintainer": {
+    "domain": "fabrikam"
+  },
+  "architect": {
+    "domain": "adventureWorks"
+  }
+}
+```
+
+### Getting the entity with the dictionary
+
+```HTTP
+GET https://graph.microsoft.com/v1.0/users/10
+```
+
+Reponse:
+
+```json
+{
+  "id": "10",
+  "displayName": "Jane Smith",
+  "roles": {
+    "author": {
+      "domain": "contoso"
+    },
+    "maintainer": {
+      "domain": "fabrikam"
+    },
+    "architect": {
+      "domain": "adventureWorks"
+    }
+  }
+}
+```
+
+### Creating an entry in the dictionary
+
+```HTTP
+POST https://graph.microsoft.com/v1.0/users/10/roles/author
+
+{
+  "domain": "contoso"
+}
+```
+
+### Updating the dictionary
+
+```HTTP
+PATCH https://graph.microsoft.com/v1.0/users/10/roles
```

Comment:
@mikepizzo can you confirm I'm using the right verb and I'm not supposed to document PUT here please?